src/core/index.ts

```diff
@@ -326,13 +354,31 @@ export class Context {
             });
         }
     }
+
+    hasFeature(feature: SDKFeature) {
+        if (!this.features) {
+            return feature === SDKFeature.PUSH ? true : false;
+        }
+
+        return this.features.includes(feature);
+    }
 }
 
 export function assertConfigValid(config: any) {
     if (typeof config !== 'object') {
         throw 'Config must be an object';
     }
 
+    const features = Array.isArray(config.features)
```

Comment:
Do we need to also consider the case where they pass an empty array, is that a valid option? (i.e. turn off all features, or do we always require at least one feature)

So can not pass, or if passed should be non-empty.

Perhaps there's a benefit to allowing the user to pass config where features are turned off if they want to progressively opt in or conduct A/B tests using their own backend.

Reply:
Thats a point, yeah the intention was purely 'no features' -> fall back case for push track, 'with features' -> then must be one or both (e.g. if dont want push then specify features and dont include it).... interesting point though, will take a look at tweaking...

Reply:
Just made the check more strict here, currently an empty array is not valid as that is in line with current impl (can revisit support for disabling all features - at this point I'm sure it will just be a case of not initialising the feature tracks at the entry point, but would need to see if that would impact / produce edge cases before implementing that support)

Reply:
Sounds good, snippet can be omitted if want to disable at present.